File: edukate-common/src/main/java/io/github/sanyavertolet/edukate/common/checks/SubmissionContext.java
package io.github.sanyavertolet.edukate.common.checks;

import lombok.Builder;
import lombok.Data;
import lombok.extern.jackson.Jacksonized;

import java.util.List;

@Data
@Builder
@Jacksonized
public class SubmissionContext {
    private String submissionId;
    private String problemId;
    private String problemText;
    private List<String> problemImageUrls;
    private List<String> submissionImageUrls;
}

File: edukate-common/src/main/kotlin/io/github/sanyavertolet/edukate/common/checks/SubmissionContext.kt
package io.github.sanyavertolet.edukate.common.checks

import io.github.sanyavertolet.edukate.common.ContentLanguage

data class SubmissionContext(
    val submissionId: Long,
    val checkResultId: Long,
    val problemId: Long,
    val problemText: String,
    val problemImageRawKeys: List<String>,
    val submissionImageRawKeys: List<String>,
    val answer: String? = null,
    val language: ContentLanguage = ContentLanguage.RU,
)
